File: packages/cmk-frontend/src/js/modules/availability.ts
/**
 * Copyright (C) 2024 Checkmk GmbH - License: GNU General Public License v2
 * This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
 * conditions defined in the file COPYING, which is part of this source code package.
 */

import {add_class, remove_class} from "./utils";

export function timeline_hover(
    timeline_nr: number,
    row_nr: number,
    onoff: number,
) {
    const row = document.getElementById(
        "timetable_" + timeline_nr + "_entry_" + row_nr,
    );
    if (!row) return;

    if (onoff) {
        add_class(row, "hilite");
    } else {
        remove_class(row, "hilite");
    }
}

export function timetable_hover(
    timeline_nr: number,
    row_nr: number,
    onoff: number,
) {
    const slice = document.getElementById(
        "timeline_" + timeline_nr + "_entry_" + row_nr,
    );
    if (!slice) return;

    if (onoff) {
        add_class(slice, "hilite");
    } else {
        remove_class(slice, "hilite");
    }
}

File: cmk/plugins/collection/graphing/availability.py
#!/usr/bin/env python3
# Copyright (C) 2024 Checkmk GmbH - License: GNU General Public License v2
# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
# conditions defined in the file COPYING, which is part of this source code package.

from cmk.graphing.v1 import graphs, metrics, perfometers, Title

UNIT_PERCENTAGE = metrics.Unit(metrics.DecimalNotation("%"))

metric_availability = metrics.Metric(
    name="availability",
    title=Title("Availability"),
    unit=UNIT_PERCENTAGE,
    color=metrics.Color.CYAN,
)

perfometer_availability = perfometers.Perfometer(
    name="availability",
    focus_range=perfometers.FocusRange(
        perfometers.Closed(0),
        perfometers.Closed(100.0),
    ),
    segments=["availability"],
)

graph_availability = graphs.Graph(
    name="availability",
    title=Title("Availability"),
    simple_lines=["availability"],
)
